cassandra-analytics-core/src/test/java/org/apache/cassandra/spark/bulkwriter/coordinatedwrite/CassandraClusterInfoGroupTest.java

```diff
@@ -0,0 +1,222 @@
+        when(clusterInfo.clusterWriteAvailability()).thenReturn(Collections.emptyMap());
+        CassandraClusterInfoGroup group = mockClusterGroup(1, index -> clusterInfo);
+        // Since there is a single clusterInfo in the group. It behaves as a simple delegation to the sole clusterInfo
+        assertThat(group.clusterWriteAvailability()).isSameAs(clusterInfo.clusterWriteAvailability());
+        assertThat(group.timeSkew(null)).isSameAs(clusterInfo.timeSkew(null));
+        assertThat(group.getLowestCassandraVersion()).isSameAs(clusterInfo.getLowestCassandraVersion());
+        assertThat(group.getTokenRangeMapping(true)).isSameAs(clusterInfo.getTokenRangeMapping(true));
+    }
+
+    @Test
+    void testAggregatePartitioner()
+    {
+        CassandraClusterInfoGroup invalidGroup = mockClusterGroup(2, index -> {
+            CassandraClusterInfo clusterInfo = mockClusterInfo("cluster" + index);
+            Partitioner partitioner = (index % 2 == 0) ? Partitioner.Murmur3Partitioner : Partitioner.RandomPartitioner;
+            when(clusterInfo.getPartitioner()).thenReturn(partitioner);
+            return clusterInfo;
+        });
+        assertThatThrownBy(invalidGroup::getPartitioner)
+        .isExactlyInstanceOf(IllegalStateException.class)
+        .hasMessage("Clusters are not running with the same partitioner kind. Found partitioners: " +
+                    "{cluster0=org.apache.cassandra.dht.Murmur3Partitioner, cluster1=org.apache.cassandra.dht.RandomPartitioner}");
+
+        CassandraClusterInfoGroup goodGroup = mockClusterGroup(2, index -> {
+            CassandraClusterInfo clusterInfo = mockClusterInfo("cluster" + index);
+            when(clusterInfo.getPartitioner()).thenReturn(Partitioner.Murmur3Partitioner);
+            return clusterInfo;
+        });
+        assertThat(goodGroup.getPartitioner()).isEqualTo(Partitioner.Murmur3Partitioner);
+    }
+
+    @Test
+    void testAggregateWriteAvailability()
+    {
+        Map<RingInstance, WriteAvailability> cluster1Availability = ImmutableMap.of(mock(RingInstance.class), WriteAvailability.AVAILABLE);
+        Map<RingInstance, WriteAvailability> cluster2Availability = ImmutableMap.of(mock(RingInstance.class), WriteAvailability.AVAILABLE);
+        CassandraClusterInfoGroup group = mockClusterGroup(2, index -> {
+            Map<RingInstance, WriteAvailability> availability = (index % 2 == 0) ? cluster1Availability : cluster2Availability;
+            CassandraClusterInfo clusterInfo = mockClusterInfo("cluster" + index);
+            when(clusterInfo.clusterWriteAvailability()).thenReturn(availability);
+            return clusterInfo;
+        });
+        assertThat(group.clusterWriteAvailability())
+        .describedAs("clusterWriteAvailability retrieved from group contains entries from both clusters")
+        .hasSize(2)
+        .containsValues(WriteAvailability.AVAILABLE, WriteAvailability.AVAILABLE);
```

Comment:
I always feel better when doing a test like this if I have 2 _different_ values that I'm combining, so maybe change one of the two clusters' ring instance to `UNAVAILABLE_DOWN` and change this assert?

Reply:
yeah.. I think I had 2 state in my mind, but somehow they are all `AVAILABLE` in the written test. Maybe copy paste error. Good call.